File: behavioral/observer/event_monitoring/typescript/event.ts
export enum EventType {
  LogInfo = "LOG_INFO",
  LogWarn = "LOG_WARN",
  LogError = "LOG_ERROR",
  LogCritical = "LOG_CRITICAL",
}

export interface IEvent {
  getType(): EventType;
  getData(): Record<string, unknown>;
  getTimestamp(): Date;
  toString(): string;
}

export class Event implements IEvent {
  private type: EventType;
  private data: Record<string, unknown>;
  private timestamp: Date;

  constructor(eventType: EventType, data: Record<string, unknown>) {
    this.type = eventType;
    this.data = data;
    this.timestamp = new Date();
  }

  public getType(): EventType {
    return this.type;
  }

  public getData(): Record<string, unknown> {
    return this.data;
  }

  public getTimestamp(): Date {
    return this.timestamp;
  }

  public toString(): string {
    const timeStr = this.timestamp
      .toISOString()
      .replace("T", " ")
      .substring(0, 19);
    return `[${timeStr}] ${this.type}: ${JSON.stringify(this.data)}`;
  }
}

File: behavioral/observer/event_monitoring/python/event.py
from dataclasses import dataclass
from datetime import datetime
from enum import Enum


class EventType(Enum):
    LOG_INFO = "INFO"
    LOG_WARN = "WARN"
    LOG_ERROR = "ERROR"
    LOG_CRITICAL = "CRITICAL"

@dataclass
class Event:
    """Data class to represent an event."""
    event_type: EventType
    data: dict
    timestamp: datetime = datetime.now()

    def __str__(self):
        return f"[{self.timestamp.strftime('%Y-%m-%d %H:%M:%S')}] {self.event_type}: {self.data}"
